<!--
  Demo launch file for villa_surface_detectors shelf_detector_action_server:
  Builds an aggregated Pointcloud using octomap and then calls the shelf_detector_action_server,
  which extracts the shelves and objects' pointcloud of a cupboard.

  rosrun villa_surface_detectors shelf_detector_action_tester #to get the action server in action

  Copy this file into your workspace and adjust as needed
  by @Arnau (Slack) @a-canya (GitHub)
-->
<launch>
    <node pkg="villa_octomap_server" type="octomap_color_server_node" name="octomap_server">
        <param name="resolution" value="0.01" />
        <param name="latch" value="false" />

        <!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
        <param name="frame_id" type="string" value="map" />

	    <param name="height_map" value="false" />
	    <param name="colored_map" value="true" />

        <!-- maximum range to integrate (speedup!) -->
        <param name="sensor_model/max_range" value="2.0" />

        <!-- data source to integrate (PointCloud2) -->
        <remap from="cloud_in" to="/hsrb/head_rgbd_sensor/depth_registered/rectified_points" />
    </node>

    <node pkg="villa_surface_detectors" type="shelf_detector_action_server" name="shelf_detector_action_server">
    </node>

</launch>
